imagehash.go

```diff
@@ -104,3 +112,86 @@ func (h *ImageHash) ToString() string {
 	}
 	return fmt.Sprintf(strFmt, kindStr, h.hash)
 }
+
+// Implement ExtImageHash from here
+func newExtImageHash(hash []uint64, kind Kind) *ExtImageHash {
```

Comment:
How about changing the name newExtImageHash -> LoadExtImageHash and use it as public API?